### PR TITLE
Feature/create progress display

### PR DESCRIPTION
**Pull request by MountainCat1**

### CODE REVIEW

File: TileTowns/Assets/Scripts/Tilemap/TileMapData.cs [hunk 1]
```diff
@@ -40,6 +42,13 @@ private void OnEnable()
         _gameState.MutationChanged += OnMutationChanged;
     }
 
+    private void OnDisable()
+    {
+        Debug.Log("XD");
```
Comment (BartekBanas) [inline TileTowns/Assets/Scripts/Tilemap/TileMapData.cs:47]: XD

File: TileTowns/Assets/Scripts/Managers/TileSelector.cs [hunk 2]
```diff
@@ -107,4 +104,6 @@ private void MoveTileHighlightTo(Vector3Int lastHighlightedCell, Vector3Int cell
         highlightGrid.SetTile(lastHighlightedCell, null);
         highlightGrid.SetTile(cell, highlightTile);
     }
+
+
```
Comment (BartekBanas) [inline TileTowns/Assets/Scripts/Managers/TileSelector.cs:108]: Kind of a weird space xD

File: TileTowns/Assets/Scripts/GameState/GameState.cs [hunk 3]
```diff
@@ -65,6 +69,8 @@ public class GameState : IGameState
     [Inject]
     public GameState(ITurnManager turnManager)
     {
+        Debug.Log("Instantiating game state");
```
Comment (BartekBanas) [inline TileTowns/Assets/Scripts/GameState/GameState.cs:72]: I'm not sure if you want to keep that log. Just making sure it's not just a forgotten artifact xD

File: TileTowns/Assets/Scripts/Buildings/Woodmill.cs [hunk 4]
```diff
@@ -15,7 +15,7 @@ public class Woodmill : Building
         public override void UpdateMutation(ITileData tileData, IGameStateTurnMutation mutation)
         {
             int adjacentForests = CountAdjacentForestTiles(tileData);
-            mutation.BuildingIncome = moneyPerWorker * tileData.WorkersAssigned * (1 + adjacentForests);
```
Comment (BartekBanas) [inline TileTowns/Assets/Scripts/Buildings/Woodmill.cs:18]: Writing this I wanted to slightly minimise the scalability. I feel like player is being punished too much whenever he doesn't maximise woodmill's output.

That could be justified that a tile is still a decent area of land so it has some trees too.

It's up to you tho

Comment (MountainCat1) [reply TileTowns/Assets/Scripts/Buildings/Woodmill.cs:18]: I would say wood mill should be a building *for* utilizing trees, and nothing else. If you want to build something on a plane field without forest wood mill should not be a way to go

File: TileTowns/Assets/Scripts/GameState/GameState.cs (hunk 3 above)
Comment (MountainCat1) [reply TileTowns/Assets/Scripts/GameState/GameState.cs:72]: Ye.... we can remove it i guess

File: TileTowns/Assets/Scripts/Buildings/Woodmill.cs (hunk 4 above)
Comment (BartekBanas) [reply TileTowns/Assets/Scripts/Buildings/Woodmill.cs:18]: And it won't be, surely farm will be significantly better withour surrounding trees

**Review by BartekBanas — APPROVED** [review]

File: TileTowns/Assets/Scripts/Buildings/Woodmill.cs (hunk 4 above)
Comment (MountainCat1) [reply TileTowns/Assets/Scripts/Buildings/Woodmill.cs:18]: to decide will add an issue

**Review by BartekBanas — DISMISSED** [review]

**Review by BartekBanas — APPROVED** [review]

**Review by BartekBanas — APPROVED** [review]

**Review by BartekBanas — APPROVED** [review]

**Pull request merged**